ledger-core-cosmos/src/cosmos/CosmosLikeAccount.cpp

```diff
@@ -80,24 +81,58 @@ namespace ledger {
                 }
 
-                // FIXME Review this!
+                static void computeAmount(CosmosLikeOperation &out, const cosmos::Message &msg) {
+                        switch (cosmos::stringToMsgType(msg.type.c_str())) {
+                                case api::CosmosLikeMsgType::MSGSEND: {
+                                        const auto& coins = boost::get<cosmos::MsgSend>(msg.content).amount;
+                                        std::for_each(coins.begin(), coins.end(), [&] (cosmos::Coin amount) {
+                                                out.amount = out.amount + BigInt::fromDecimal(amount.amount);
+                                        });
+                                } break;
+                                case api::CosmosLikeMsgType::MSGDELEGATE: {
+                                        out.amount = boost::get<cosmos::MsgDelegate>(msg.content).amount.amount;
+                                } break;
+                                case api::CosmosLikeMsgType::MSGUNDELEGATE: {
+                                        out.amount = boost::get<cosmos::MsgUndelegate>(msg.content).amount.amount;
+                                } break;
+                                case api::CosmosLikeMsgType::MSGREDELEGATE: {
+                                        out.amount = boost::get<cosmos::MsgRedelegate>(msg.content).amount.amount;
+                                } break;
+                                case api::CosmosLikeMsgType::MSGSUBMITPROPOSAL: {
+                                        const auto& coins = boost::get<cosmos::MsgSubmitProposal>(msg.content).initialDeposit;
+                                        std::for_each(coins.begin(), coins.end(), [&] (cosmos::Coin amount) {
+                                                out.amount = out.amount + BigInt::fromDecimal(amount.amount);
+                                        });
+                                } break;
+                                case api::CosmosLikeMsgType::MSGVOTE: {
+                                        // No amount-like data for this type of operation
+                                } break;
+                                case api::CosmosLikeMsgType::MSGDEPOSIT: {
+                                        const auto& coins = boost::get<cosmos::MsgDeposit>(msg.content).amount;
+                                        std::for_each(coins.begin(), coins.end(), [&] (cosmos::Coin amount) {
+                                                out.amount = out.amount + BigInt::fromDecimal(amount.amount);
+                                        });
+                                } break;
+                                case api::CosmosLikeMsgType::MSGWITHDRAWDELEGATIONREWARD: {
+                                        // No amount-like data for this type of operation
+                                } break;
+                                default: {
+                                } break;
```

Comment:
You can fuse the `default`, `MSGVOTE` and the `MSGWITHDRAWDELEGATIONREWARD` to make it more obvious that no amount is expected in those cases